src/GovUk.Education.ExploreEducationStatistics.Content.Services/ReleaseService.cs

```diff
@@ -24,37 +26,76 @@ public class ReleaseService : IReleaseService
+        public async Task<Either<ActionResult, ReleaseViewModel>> Get(string publicationSlug, string? releaseSlug = null)
         {
-            return await CreateFromCachedPublicationAndRelease(publicationPath, releasePath,
-                    (publication, release) => new ReleaseViewModel(release, publication))
-                .OnSuccessCombineWith(model => _methodologyService.GetSummariesByPublication(model.Publication.Id))
-                .OnSuccess(viewModelAndMethodologies =>
+            return await _publicationService.Get(publicationSlug)
+                .OnSuccessCombineWith(publication => _methodologyService.GetSummariesByPublication(publication.Id))
+                .OnSuccess(async tuple =>
                 {
-                    var (viewModel, methodologies) = viewModelAndMethodologies;
-                    viewModel.Publication.Methodologies = methodologies;
-                    return viewModel;
+                    var (publication, methodologies) = tuple;
+                    var cachedRelease = await GetCachedRelease(publicationSlug, releaseSlug);
+
+                    if (cachedRelease.IsRight
+                        && cachedRelease.Right is not null
+                        && publication is not null)
+                    {
+                        var result = new Either<ActionResult, ReleaseViewModel>(new ReleaseViewModel(
+                            _mapper.Map<CachedReleaseViewModel>(cachedRelease.Right),
+                            _mapper.Map<PublicationViewModel>(publication)
+                        ));
+
+                        result.Right.Publication.Methodologies = methodologies;
+                        return result;
+                    }
+
+                    return new NotFoundResult();
                 });
         }
 
-        public async Task<Either<ActionResult, ReleaseSummaryViewModel>> GetSummary(string publicationPath,
-            string releasePath)
+        public async Task<Either<ActionResult, ReleaseSummaryViewModel>> GetSummary(string publicationSlug,
+            string? releaseSlug)
         {
-            return await CreateFromCachedPublicationAndRelease(publicationPath, releasePath,
-                (publication, release) => new ReleaseSummaryViewModel(release, publication));
+            return await _persistenceHelper
+                .CheckEntityExists<Publication>(query => query
+                    .Where(p => p.Slug == publicationSlug))
+                .OnSuccess(async _ =>
+                {
+                    var publication = await _publicationService.Get(publicationSlug);
+                    var release = await GetCachedRelease(publicationSlug, releaseSlug);
+
+                    if (release.IsRight
+                        && release.Right is not null
+                        && publication.IsRight
+                        && publication.Right is not null)
```

Comment:
Isn't `_publicationService.Get` going to mean the `.CheckEntityExists` is a bit redundant here? Same comment as before that you might want to consider using `OnSuccessCombineWith`

Reply:
Good point. Removed `CheckEntityExists`